src/gui/dialogs/logging.cpp

```diff
@@ -0,0 +1,118 @@
+/*
+ Part of the Battle for Wesnoth Project http://www.wesnoth.org/
+
+ This program is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2 of the License, or
+ (at your option) any later version.
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY.
+
+ See the COPYING file for more details.
+ */
+
+#define GETTEXT_DOMAIN "wesnoth-lib"
+
+#include "gui/dialogs/logging.hpp"
+
+#include "gui/auxiliary/find_widget.hpp"
+#include "gui/widgets/grid.hpp"
+#include "gui/widgets/listbox.hpp"
+#include "gui/widgets/settings.hpp"
+#include "gui/widgets/toggle_button.hpp"
+#include "gui/widgets/window.hpp"
+
+#include "log.hpp"
+
+namespace gui2
+{
+
+REGISTER_DIALOG(logging)
+
+tlogging::tlogging()
+{
+	//list of names must match those in logging.cfg
+	widget_id_.push_back("err");
+	widget_id_.push_back("warn");
+	widget_id_.push_back("info");
+	widget_id_.push_back("debug");
+
+	domain_list_end_ = "the end";
+
+	//empty string is the filter (in other words, this grabs the whole list of domains)
+	std::string temp_string = lg::list_logdomains("");
+	//std::cout<<temp_string; //use to print the full log domain list
+	std::string one_domain;
+
+	std::istringstream iss(temp_string, std::istringstream::in);
+
+	while(iss >> one_domain){
+		domain_list_.push_back(one_domain);
+	}
+	domain_list_.push_back(domain_list_end_);
+}
+
+void tlogging::pre_show(twindow& window)
+{
+	set_restore(true); //why is this done manually?
+
+	tlistbox& logger_box = find_widget<tlistbox>(&window, "logger_listbox", false);
+
+	for(unsigned int counter = 0; domain_list_[counter] != domain_list_end_; counter++){
```

Comment:
I assume this is what @gfgtdf is referring to... look to where you use it, it does seem very weird. To begin with, assuming you actually want to iterate this way, this should probably be a const string in the source file and not exist at all in the header. Furthermore, this isn't really a good way to iterate through the logdomains.The `domain_list_` is a vector, so you can query its `.size()`, or if you don't need the counter for anything other than the line just below this, you can just use the range-for iteration syntax:

```c++
for(std::string this_domain : domain_list_)
```

That would replace this line and the next line, and eliminate the need for this weird `domain_list_end_` variable.